### ISSUE TITLE
How to disable this user warning?

### ISSUE THREAD

**Issue by jinfagang** [ZhaoJ9014/face.evoLVe.PyTorch#4]
When run detector to show result on detect result, there is an warning:

```
/align/get_nets.py:70: UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.
  a = F.softmax(a)
detector.py:82: UserWarning: volatile was removed and now has no effect. Use `with torch.no_grad():` instead.
  img_boxes = Variable(torch.FloatTensor(img_boxes), volatile = True)

```

**Comment by ZhaoJ9014** [comment]
Hi, thank you for your question. We did not set the specific dimension for "F.softmax" at Line 70 in /align/get_nets.py, which has been deprecated. You can optionally include a "dim=X" ("X" is the output neuron number) to avoid this warning. For the second warning, you can remove ", volatile = True" from "img_boxes = Variable(torch.FloatTensor(img_boxes), volatile = True)" at Line 82 in /align/detector.py and put the relevant code within "with torch.no_grad():". Hope this helps:)

**Comment by jinfagang** [comment]
@ZhaoJ9014 Thanks for your dedicate reply. BTW, how much MTCNN fps runs on CPU for now? Haven't test on a video but image seems fast. And is there any training process included to train a MTCNN face detector?

**Comment by ZhaoJ9014** [comment]
Welcome. 1) We haven't performed video test on CPU, but the speed should be also satisfactory. 2) Currently, we did not include the training process due to a common issue that the re-implementations can hardly reproduce the results reported by the MTCNN original paper. We thus choose to transfer the weights and implement the inference process. Please kindly inform us if you get any new results by performing video test on CPU or reproducing the reported results with self-training in the future.

**Comment by watson21** [comment]
> Hi, thank you for your question. We did not set the specific dimension for "F.softmax" at Line 70 in /align/get_nets.py, which has been deprecated. You can optionally include a "dim=X" ("X" is the output neuron number) to avoid this warning.

specifically in this case, how many output neuron number of this model.
I using Onet, Pnet and Rnet
Btw, i don't see any flatten layer or linear layer in Onet, Pnet or Rnet
Thanks you
